### PR TITLE
Correcting gradient accumulation in the training section

### PR DESCRIPTION
**Pull request by timlautk**
Putting the forward pass within the no_sync() context along with the backward pass; otherwise gradients will still be synchronized. For details, check this from the PyTorch repo: https://github.com/pytorch/pytorch/blob/main/torch/nn/parallel/distributed.py#L1427

### CODE REVIEW

**Comment by KellerJordan** [comment]
this might be slightly slower due to moving next_batch after the backward. not sure will have to check

**Comment by timlautk** [comment]
You're right. I have check that. Just putting that back before the backward.
